src/bcccp/tickets/season/SeasonTicketDAO.java

```diff
@@ -3,64 +3,108 @@
 import bcccp.tickets.season.ISeasonTicket;
 import bcccp.tickets.season.IUsageRecordFactory;
 
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Date;
+
 public class SeasonTicketDAO implements ISeasonTicketDAO {
 
+	private List<ISeasonTicket> seasonTickets;
 	private IUsageRecordFactory factory;
 
 	
 	
 	public SeasonTicketDAO(IUsageRecordFactory factory) {
-		//TOD Implement constructor
+
+		this.factory = factory;
+
+		seasonTickets = new ArrayList<>();
+
 	}
 
 
 
 	@Override
 	public void registerTicket(ISeasonTicket ticket) {
-		// TODO Auto-generated method stub
+
+		seasonTickets.add(ticket);
 		
 	}
 
 
-
 	@Override
 	public void deregisterTicket(ISeasonTicket ticket) {
-		// TODO Auto-generated method stub
+
+		// 'deregister' means: remove ticket from the seasonTickets ArrayList
+
+		Iterator<ISeasonTicket> sTicketRecs = seasonTickets.iterator();
+
+		while (sTicketRecs.hasNext()) {
+
+			if (sTicketRecs.next().getId().equals(ticket.getId())) {
+
+				sTicketRecs.remove();
+			}
+		}
 		
 	}
 
 
-
 	@Override
 	public int getNumberOfTickets() {
-		// TODO Auto-generated method stub
-		return 0;
-	}
 
+		return seasonTickets.size();
+	}
 
 
 	@Override
 	public ISeasonTicket findTicketById(String ticketId) {
-		// TODO Auto-generated method stub
-		return null;
-	}
 
+		Iterator<ISeasonTicket> sTicketRecs = seasonTickets.iterator();
+
+		ISeasonTicket sTicket = null;
+
+		while (sTicketRecs.hasNext()) {
+
+			if (sTicketRecs.next().getId().equals(ticketId)) {
+
+				sTicket = sTicketRecs.next();
+			}
+			else {
+				sTicket = null;
+			}
+		}
+		return sTicket;
+	}
 
 
 	@Override
 	public void recordTicketEntry(String ticketId) {
-		// TODO Auto-generated method stub
-		
-	}
 
+		// This method creates a new usage record with current day and time as the startTime
+		// and uses recordUsage method from SeasonTicket class to record it to the ArrayList
+
+		Date dateTime = new Date();
+
+		IUsageRecord usageRecord = factory.make(ticketId, dateTime.getTime());
+
+		findTicketById(ticketId).recordUsage(usageRecord);
+
+	}
 
 
 	@Override
 	public void recordTicketExit(String ticketId) {
-		// TODO Auto-generated method stub
+
+		// Finds an existing usage record and records the current day and time (on exiting of vehicle)
+		// on the record
+
```

Comment:
9. TYPE. Formatting: Google style Infringement
   Each time a new block or block-like construct is opened, the indent increases by two spaces. This defect applies to the 
   document as whole which appears to open with an indent of 4 spaces.